tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -0,0 +1,246 @@
+                            GMM,
+                            classdef.TectType[tect_type],
+                            rupture_context_df,
+                            str(im),
+                            psa_periods if im is gc.im.IMType.pSA else None,
+                        )
+                        # Matching the index with rupture_df
+                        # to have a right rupture label
+                        gmm_calculated_df.set_index(
+                            rupture_df[
+                                rupture_df["rupture_name"].isin(
+                                    rupture_context_df["rupture_name"]
+                                )
+                            ].index,
+                            inplace=True,
+                        )
+                        gmm_calculated_df.columns = [
+                            f"{'_'.join(col.split('_')[:2]) if im is gc.im.IMType.pSA else im}_sigma"
+                            if col.endswith("std_Total")
+                            else col
+                            for col in gmm_calculated_df
+                        ]
+
+                        # Write an im_df to the given station/site
+                        imdb_dict[db_type].open()
+                        imdb_dict[db_type].add_im_data(
+                            site,
+                            gmm_calculated_df.loc[
+                                :,
+                                gmm_calculated_df.columns.str.endswith(
+                                    ("mean", "sigma")
+                                ),
+                            ],
+                        )
+                        imdb_dict[db_type].close()
+
+    print("Writing metadata")
+    common.write_metadata_and_close(
+        imdb_dict,
+        background_sources_ffp,
+        rupture_df,
+        site_df,
+        vs30_ffp,
+        psa_periods,
+        ims,
+        model_dict_ffp,
+    )
+    print("Done")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("background_txt", help="background txt file")
+    parser.add_argument("site_source_db")
+    parser.add_argument("vs30_file")
+    parser.add_argument("output_dir")
+    parser.add_argument(
+        "--z-file", help="File name of the Z data",
+    )
+    parser.add_argument(
+        "--periods",
+        default=common.PERIOD,
+        nargs="+",
+        help="Which pSA periods to calculate for",
+    )
+    parser.add_argument(
+        "--im",
+        default=common.IM_TYPE_LIST,
+        nargs="+",
+        help="Which IMs to calculate",
+        type=gc.im.IMType,
+    )
+    parser.add_argument(
+        "--model-dict",
+        help="model dictionary to specify which model to use for each tect-type",
+    )
+    parser.add_argument(
+        "--suffix", "-s", help="suffix for the end of the imdb files", default=None,
+    )
+
+    return parser.parse_args()
+
+
+def calculate_emp_ds():
```

Comment:
Don't think this function is needed, just put that code under `if __name__ == "__main__"` directly?

Reply:
Ahhh, the old Claudio vs Jason style. I prefer the current way so it doesn't chuck a bunch of stuff in the global namespace. But Claudio prefers the other way. And we still haven't came to an agreement. :toilet: